crates/ra_hir/src/ty/primitive.rs

```diff
@@ -38,12 +48,12 @@ pub enum UncertainFloatTy {
 }
 
 impl UncertainFloatTy {
-    pub fn from_name(name: &Name) -> Option<UncertainFloatTy> {
-        if let Some(ty) = FloatTy::from_name(name) {
-            Some(UncertainFloatTy::Known(ty))
-        } else {
-            None
-        }
+    pub(crate) fn from_type_name(name: &Name) -> Option<UncertainFloatTy> {
+        FloatTy::from_type_name(name).map(UncertainFloatTy::Known)
+    }
+
+    pub(crate) fn from_suffix(suffix: &str) -> Option<UncertainFloatTy> {
```

Comment:
In terms of code, these two methods are basically exact duplicates. From the philosophical POV, they are different though: suffix in `92i32` seems to be just a simple string, while in `let _: i32 = 92` `i32` seems to be a path, subject to hygiene.